### PR TITLE
MIN-270: Stratum v1 Rust no_std no_alloc Client

### PR DESCRIPTION
**Pull request by georgesFoundation**
Add a no_std no_alloc Stratum v1 client crate

### CODE REVIEW

**Comment by georgesFoundation** [comment]
Note, I still can't `cargo package` it until faster-hex merge and publish nervosnetwork/faster-hex/pull/48 and nervosnetwork/faster-hex/pull/49

File: Cargo.toml [hunk 1]
```diff
@@ -28,17 +29,23 @@ bs58 = "0.5"
 clap = { version = "4", features = ["cargo"] }
 crc = "3"
 criterion = { version = "0.4" }
+defmt = "0.3"
+derive_more = { version = "1.0.0-beta", default-features = false }
```
Comment (jeandudey) [inline Cargo.toml:33]: I think the `1.0.0` version is out of beta now: https://crates.io/crates/derive_more/1.0.0

File: stratum-v1/src/client/request.rs [hunk 2]
```diff
@@ -0,0 +1,369 @@
...
+    #[serde(skip_serializing_if = "Option::is_none")]
+    ///An identifier established by the Client.
+    ///
+    ///If not present, request is notification to which
+    ///there should be no response.
+    pub id: Option<u64>,
+    ///A String containing the name of the method to be invoked
+    ///
+    ///By default is static buffer of 32 bytes.
+    pub method: String<23>,
```
Comment (jeandudey) [inline stratum-v1/src/client/request.rs:41]: I think this is a typo? e.g. 23 -> 32?

**Review by jeandudey — CHANGES_REQUESTED** [review]
Nice! LGTM, just some nit picks

File: Cargo.toml (hunk 1 above)
Comment (georgesFoundation) [reply Cargo.toml:33]: great !

File: stratum-v1/src/client/request.rs (hunk 2 above)
Comment (georgesFoundation) [reply stratum-v1/src/client/request.rs:41]: good catch !

**Review by jeandudey — APPROVED** [review]
LGTM!

**Pull request merged**